pkg/util/validation/limits.go

```diff
@@ -1335,6 +1335,24 @@ func SmallestPositiveNonZeroDurationPerTenant(tenantIDs []string, f func(string)
 	return *result
 }
 
+// LargestPositiveNonZeroDurationPerTenant is returning the maximum positive
+// and non-zero value of the supplied limit function for all given tenants. In
+// many limits a value of 0 means unlimited so the method will return 0 only if
+// all inputs have a limit of 0 or an empty tenant list is given.
+func LargestPositiveNonZeroDurationPerTenant(tenantIDs []string, f func(string) time.Duration) time.Duration {
+	var result *time.Duration
```

Comment:
nit: Using `nil` and a pointer for this instead of a `0` seems a little odd

Reply:
Good point. Simplified in https://github.com/grafana/mimir/pull/11069/commits/3ee482ff859cbf71e95bc01a7aa628dad31ee9b5